library/Icinga/Crypt/AesCrypt.php

```diff
@@ -50,22 +60,27 @@ class AesCrypt
     /** @var string The authentication tag which is passed by reference when using AEAD cipher mode */
     private $tag;
 
-    /** @var string The cipher method */
-    private $method = 'AES-128-GCM';
+    private $method;
 
     const GCM_SUPPORT_VERSION = '7.1';
 
-    public function __construct($random_bytes_len = 128)
+    public function __construct()
     {
-        if (version_compare(PHP_VERSION, self::GCM_SUPPORT_VERSION, '<')) {
-            $this->method = 'AES-128-CBC';
+        foreach (self::METHODS as $key => $method) {
+            if ($key === 0 && version_compare(PHP_VERSION, self::GCM_SUPPORT_VERSION, '<')) {
+                continue;
+            }
+            if (in_array($method, openssl_get_cipher_methods())) {
+                $this->method = $method;
+                break;
+            }
         }
 
-        $this->key = random_bytes($random_bytes_len);
+        $this->key = random_bytes(128);
```

Comment:
Why is this constant now? I see this change problematic, since it's been previously a parameter that may have already been used. (Backwards incompatible change)

Is this really necessary?

Reply:
I wanted to make sure that the user does not extend the length because the rememberme table cannot handle it if the length is greater than 128.

It's not really necessary.

Reply:
It's the RememberMe class which accesses the database. AesCrypt is a utility class which doesn't impose storage in the database :wink: